worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/XmlConverter.java

```diff
@@ -171,16 +166,23 @@ private static void appendEmailHeaderToBuilder(StringBuilder headersBuilder, Str
         if(headerValues.isEmpty()){
             LOG.info("No '" + headerSourceFieldName + "' values to add in email headers during markup.");
         }
-        else if (headerValues.size() > 1) {
-            throw new AddHeadersException("Unable to add email headers. Multiple field entries found for header: "
-                    +headerSourceFieldName);
-        }
         else{
+            boolean isPriorityHeader = headerSourceFieldName.equals("priority");
+
             headersBuilder.append(headerStartText+": ");
-            String headerValue = headerSourceFieldName.equals("priority")
-                    ? Normalizations.normalizePriority(headerValues.get(0))
-                    : headerValues.get(0);
+            Iterator headersIterator = headerValues.iterator();
+            String headerValue = isPriorityHeader == true
```

Comment:
I favored allowing multiple priority fields since I cannot state for certain there is no scenario where multiple priority fields aren't passed on the document. We already made an assumption for multiple values on all headers so wary of doing so again.

Restructured the code so that if this is for the priority header then the normalization occurs during the initial stream mapping.